src/renderers/dom/ReactDOM.js

```diff
@@ -133,6 +133,13 @@ if (__DEV__) {
       }
     }
   }
+
+  // Check if using Polymer and we are not using native shadow
+  var shadyDom = window.Polymer !== undefined && !window.Polymer.useNativeShadow;
+  warning(
+    !shadyDom,
+    'It looks like you are using the shady-dom. Careful things might break!'
```

Comment:
Can we change "Careful things might break" to something a little more formal / explanative? Like maybe "Using shady-dom with React may cause subtle issues and is not recommended" @gaearon any input on phrasing here?